### PR TITLE
Support biblatex apa citation for legal entry types

### LINKED ISSUE #8931: Support APA Legal citation types
**Is your suggestion for improvement related to a problem? Please describe.**

The BibLaTeX manual gives a number of "unsupported" styles of specific names, recognized but treated as `@misc`.  Certain styles, particularly biblatex-apa, support these.  In particular, the legal styles such as `jurisdiction` (court cases), `legislation`, etc.

Of particular note is the [style example for jurisdiction](https://github.com/plk/biblatex-apa/blob/master/bibtex/bib/biblatex-apa-test-references.bib#L2541-L2550) provided by biblatex-apa:

```
@JURISDICTION{11.4:5,
  TITLE                = {Durflinger v. Artiles},
  CITATION             = {563 F.Supp. 332 and aff'd and 727 F.2d 888},
  CITATION+an:citeorg  = {1="D. Kan.";3="10th Cir."},
  CITATION+an:citedate = {1="1981";3="1984"},
  CITATION+an:citeinfo = {2=appeal},
  URL                  = {https://openjurist.org/727/f2d/888/durflinger-v-artiles},
  DATE                 = {1984},
  ORIGDATE             = {1981}
}
```

These include *annotation* fields, `CITATION+an:...`

**Describe the solution you'd like**

Inclusion of style types for these such styles BibLaTeX treats as `@misc`, complete with the additional annotations, preferably under the `Annotation` tab.

It would be a larger change to also include automatic citation, for example `Legislation` based on State and code reference or bill references, or Court cases based on jurisdiction and case number, etc., as this would entail pulling information from many, *many* sources, all with different presentation.  The scope of this issue is merely providing the style type for selection by the user, who will fill in the details on their own, and excludes any such automated reference generation.

### PR DESCRIPTION
**Pull request by Ognimalf**
fixes #8931

Introduces multiple entry types that are based on the BibLaTeX standard.

grafik

grafik

grafik

grafik

grafik

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by calixtus** [comment]
Hi @Ognimalf, thanks for your interest in JabRef and your addition. However, I think your additional entry type is misplaced in BibTexEntryTypeDefinitions.java, since we exclusivly implement in that class standard entry types as defined by the documentation (https://www.ctan.org/tex-archive/biblio/bibtex/contrib/doc/). Instead I would ask you to implement it like BiblatexSoftwareEntryTypeDefinitions.java as another class of custom entry types, maybe something called 'precustomized'?

**Comment by Siedlerchr** [comment]
Please have a look at the checkstyle/reviewdog issues. Import JabRef's code style https://github.com/JabRef/jabref/tree/main/config

File: src/main/java/org/jabref/model/entry/types/StandardEntryType.java [hunk 1]
```diff
@@ -21,7 +21,11 @@ public enum StandardEntryType implements EntryType {
     Unpublished("Unpublished"),
     // Biblatex
     BookInBook("BookInBook"),
+    CONSTITUTION("Constitution"),
```
[the remark this answers is not in the document]
Comment (Siedlerchr) [reply src/main/java/org/jabref/model/entry/types/StandardEntryType.java:24]: @Ognimalf  Overlooked that yesterday. These fields should go into their own class. Please create a new class for these: see for exmaple IEEE Field . The biblatex software should have been implemented as an own class as well, but think we didn't saw this
https://github.com/JabRef/jabref/blob/bb011c9313367a28990ae213b3920fe6cd10d1dc/src/main/java/org/jabref/model/entry/field/IEEEField.java

File: src/main/java/org/jabref/model/entry/field/FieldFactory.java [hunk 2]
```diff
@@ -73,13 +73,23 @@ public static String serializeFieldsList(Collection<Field> fields) {
                      .collect(Collectors.joining(DELIMITER));
     }
 
+    public static <T> Field parseField(T type, String fieldName) {
```
Comment (koppor) [inline src/main/java/org/jabref/model/entry/field/FieldFactory.java:76]: 💕🎉

**Pull request merged**